api/healthcheck/healthcheck_controller.go

```diff
@@ -17,32 +17,36 @@
 package healthcheck
 
 import (
+	h "github.com/InVisionApp/go-health"
+	"github.com/Peripli/service-manager/pkg/health"
 	"github.com/Peripli/service-manager/pkg/util"
 	"net/http"
 
-	"github.com/Peripli/service-manager/pkg/health"
 	"github.com/Peripli/service-manager/pkg/log"
 	"github.com/Peripli/service-manager/pkg/web"
 )
 
 // controller platform controller
 type controller struct {
-	indicator health.Indicator
+	health    h.IHealth
+	tresholds map[string]int64
 }
 
-// NewController returns a new healthcheck controller with the given indicators and aggregation policy
-func NewController(indicators []health.Indicator, aggregator health.AggregationPolicy) web.Controller {
+// NewController returns a new healthcheck controller with the given health and tresholds
+func NewController(health h.IHealth, tresholds map[string]int64) web.Controller {
```

Comment:
NewController is a public api that will be used in other projects- so you have two options - go for a stable api (pass settings and know that when new settings are added the controller public api wont change) or pass minimal and risk having to refactor a public api afterwards. Both approaches are acceptable.